types/node-rsa/index.d.ts

```diff
@@ -0,0 +1,197 @@
+// Type definitions for node-rsa 0.4
+// Project: https://github.com/rzcoder/node-rsa
+// Definitions by: Ali Taheri <https://github.com/alitaheri>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/// <reference types="node" />
+
+declare class NodeRSA {
+    /**
+     * Generate new key with length specified.
+     */
+    constructor(key?: NodeRSA.KeyBits);
```

Comment:
Well it's how it's documented. There isn't any other way to properly type this.

it either takes an object of form `{ b: 512}` that generates a new key. Or the form below which loads a key from string/ buffer or key components with additional arguments about the key. the first form shouldn't take any other arguments.